mealie/core/settings/settings.py

```diff
@@ -34,6 +35,27 @@ def determine_secrets(data_dir: Path, production: bool) -> str:
         return new_secret
 
 
+def get_secrets_dir() -> str | None:
+    """
+    Returns a directory to load secret settings from, or `None` if the secrets
+    directory does not exist or cannot be accessed.
+    """
+    secrets_dir = "/run/secrets"
+
+    # Check that the secrets directory exists.
+    if not os.path.exists(secrets_dir):
+        print(f"Secrets directory '{secrets_dir}' does not exist")
```

Comment:
I'm struggling to figure out how to get access to the `logging` lib here, as I'm unable to pass `self` to `get_secrets_dir` as when this is called, `AppSettings` is not instantiated yet.

Reply:
I think we'll either need to accept that `print` is OK here, or modify `get_logging` to have an extra argument to not try and call `get_app_settings` when `__root_logger` is `None`.

https://github.com/mealie-recipes/mealie/blob/37d93d4e4b255b01aab4ea7258058c2fc818426a/mealie/core/root_logger.py#L20-L21

Reply:
Can we split app settings into two parts? Something like:

```python
class AppLoggingSettings(BaseSettings):
    """Subset of AppSettings to only access logging-related settings"""

    TESTING: bool = False
    PRODUCTION: bool

    LOG_CONFIG_OVERRIDE: Path | None = None
    """path to custom logging configuration file"""
    LOG_LEVEL: str = "info"
    """corresponds to standard Python log levels"""


class AppSettings(AppLoggingSettings):
    theme: Theme = Theme()

    BASE_URL: str = "http://localhost:8080"
    ...  # everything else here
```

Then add a new method `get_logging_settings`, and just copy what we have with `get_app_settings`, etc.

Reply:
Thanks! This helped, as did calling `get_secrets_dir` outside of `AppSettings.__init__` by passing `_secrets_dir` as an argument.